### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by hyc17003**
Within this branch, the following were done:
- Update DG to include find command implementation
- Update DG use case for find command in accordance with what the team had decided
- Standardise some formatting inconsistencies

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -394,6 +394,57 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+<puml src="diagrams/FindSequenceDiagram.puml" width="400" />
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+<puml src="diagrams/FindActivityDiagram.puml" width="500" />
+
+#### Rationale
+
+- **Flexibility**: The `FindCommand` provides flexibility to users by allowing them to choose whether to edit a person by name or NRIC, whichever is faster or available.
```
Comment (kanna-1) [inline docs/DeveloperGuide.md:423]: Typo here. Should be `find` instead of `edit`.

Comment (nubnubyas) [inline docs/DeveloperGuide.md:423]: FindCommand should allow users to choose whether to find a person by name or NRIC, not edit a person.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -394,6 +394,57 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+- **Usage**: Users invoke the `FindCommand` by specifying the command word, followed by the name or NRIC of the person(s) they wish to find.
+    - The command format is: `find n/NAME` or `find id/IC_NUMBER`.
+- **`execute` method**: The `FindCommand` executes the search by using the specified predicates (`NameContainsKeywordsPredicate` or `IdContainsKeywordsPredicate`) to filter and list all persons matching the search criteria.
+- **Validation**: The `FindCommand` performs validation to ensure at least one keyword is provided. It searches based on either name or NRIC, to speed up the search and prevent possible conflicts if name and NRIC do not match each other.
+- **Execution**: When executed, the `FindCommand` identifies the person(s) being searched for based on the provided name or NRIC. If a name is provided as keyword, a `FindCommand(NameContainsKeywordsPredicate)` is created, and if an NRIC is provided as keyword, a `FindCommand(IdContainsKeywordsPredicate)` is created. `updateFilteredPersonList` will then update the filter of the filtered person list to filter by the given name or NRIC predicate (keyword).
+
```
Comment (kanna-1) [inline docs/DeveloperGuide.md:412]: Can I confirm that FindCommand is an overloaded constructor depending on whether Name or Nric is provided to the constructor?

Comment (hyc17003) [reply docs/DeveloperGuide.md:412]: Yep that's correct

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (hyc17003) [reply docs/DeveloperGuide.md]: Thank you! I've fixed it

Comment (hyc17003) [reply docs/DeveloperGuide.md]: Thank you! I've fixed it

**Comment by kanna-1** [comment]
Ok LGTM. Merging

**Pull request merged**